src/core/middleware/requestId.js

```diff
@@ -0,0 +1,21 @@
+/* @flow */
+import httpContext from 'universal-express-http-context';
+import uuidv4 from 'uuid/v4';
+import type { $Request, $Response, NextFunction, Middleware } from 'express';
+
+import { AMO_REQUEST_ID_HEADER } from 'core/constants';
+
+const requestId = (
+  req: $Request,
+  res: $Response,
+  next: NextFunction,
+  { _httpContext = httpContext }: {| _httpContext: typeof httpContext |} = {},
+): Middleware => {
+  const amoRequestId = req.headers[AMO_REQUEST_ID_HEADER] || uuidv4();
+
+  _httpContext.set(AMO_REQUEST_ID_HEADER, amoRequestId);
+  res.setHeader(AMO_REQUEST_ID_HEADER, amoRequestId);
```

Comment:
I think it is, it should not be passed to Google, but users can read it. That'd be a useful way to debug an issue.

Reply:
Maybe ask ulfr again? I'm not sure he understood that we are exposing it in a response header. I definitely think it will be useful but if there is a privacy concern we may need to keep it disabled in prod until we need to debug something specific.

Reply:
@jvehent re: request/correlation ID on AMO–can we expose this unique ID (generated for every request on AMO) to the response sent to the user (in a header)?

Reply:
I initially understood that this ID would be passed between internal services, not back to the user. That said, I do think it's still fine to send it back to the user as a response header. We have session cookies that do more tracking than that and are ok.

Reply:
Initially it was an internal-only ID but exposing it eases our life. Thanks!